core/offchain/src/lib.rs

```diff
@@ -58,6 +60,7 @@ pub struct OffchainWorkers<Client, Storage, Block: traits::Block> {
 	client: Arc<Client>,
 	db: Storage,
 	_block: PhantomData<Block>,
+	thread_pool: Mutex<ThreadPool>,
```

Comment:
The `Mutex` isn't needed.

Reply:
`ThreadPool` isn't Send/Sync

Reply:
https://github.com/rust-threadpool/rust-threadpool/blob/21a70c7e8b19fb73a9f3f44a55ca1b7333d804f9/src/lib.rs#L1212-L1228

I don't see a test for `Sync` for `ThreadPool`, but I would be surprised if it wasn't `Sync` as well.

Reply:
https://github.com/rust-threadpool/rust-threadpool/issues/96

There is an issue for this.